src/views/Home.jsx

```diff
@@ -1,12 +1,76 @@
 import './Home.css';
 import { SingleList } from '../components/SingleList.jsx';
+import { useState, useEffect } from 'react';
+import { createList } from '../api/firebase.js';
+import { useNavigate } from 'react-router-dom';
+
+export function Home({ data, userId, userEmail, setListPath }) {
+	const [newListName, setNewListName] = useState('');
+	const [newListSuccess, setNewListSuccess] = useState(null);
+	const [message, setMessage] = useState(null);
```

Comment:
I feel the `message` state is not necessary in this case. The purpose of this state is to store the message to show an alert, and the message will only show two possibilities:  `List created successfully.` or `List could not be created.` You can just call directly in the `alert('')`

The`useEffect` and `newListSuccess` are not required as well in this case. We can just call the alert directly in the `handleSubmit` function

`handleSubmit` function will look like this.
```jsx
	const handleSubmit = async (e) => {
		e.preventDefault();
		const currentLists = data.map((list) => {
			return list.name;
		});

		//Check if the new list has a title that already exists in db
		if (currentLists.indexOf(newListName) > -1) {
			alert('List already exists. Please enter a unique list name.');
		} else {
			//Create new list
			const response = await createList(userId, userEmail, newListName);

			if (response) {
				alert('List created successfully.');
				navigate('/list');
				await setListPath(`${userId}/${newListName}`);
			} else {
				alert('List could not be created.');
			}
		}

		setNewListName('');
	};
```

Let me know if you need more explanation, I would be happy to explain more. Sorry for the long feedback.

Reply:
Thanks, this makes sense! I thought we had tried it with the setListPath and navigate inside the handleSubmit function and we had problems with getting the list path to update and navigate properly so we had tried a few other things including the useEffect. But I just tried this and it works so I just pushed the changes--thanks for the suggestion!